### PR TITLE
Upgrade to nodejs v18

### PR DESCRIPTION
**Pull request by legobeat**
* v12 is EOL since 2022-04-30
* v14 will reach EOL 2023-04-30
* v18 LTS 2022-10-25 - 2023-10-23

https://github.com/nodejs/release#release-schedule

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v14
+v18
```
Comment (FrederikBolding) [inline .nvmrc:1]: I think we should stick to Node 16 until the extension upgrades to 18.

Comment (legobeat) [reply .nvmrc:1]: Fair enough; will address!

**Comment by Gudahtt** [comment]
Do we want to stop supporting Node.js v14 in our libraries? Lots of them are used by other projects outside of MetaMask. We've been tracking the maintenance LTS schedule (roughly) as a courtesy to other users of our libraries. That schedule would have us supporting v14 until next year.

**Comment by legobeat** [comment]
> Do we want to stop supporting Node.js v14 in our libraries? Lots of them are used by other projects outside of MetaMask. We've been tracking the maintenance LTS schedule (roughly) as a courtesy to other users of our libraries. That schedule would have us supporting v14 until next year.

With that in light, let's add 14 back to the ci matrix.

**Comment by Gudahtt** [comment]
Sounds good! Totally open to dropping v14 if it benefits us enough to be worth the tradeoff of slightly inconveniencing other projects. Just wanted to highlight that as a tradeoff.

**Comment by legobeat** [comment]
I'm agnostic. Either way, I think breaking existing expectations needs some prior announcement and should be a separate change.

**Comment by legobeat** [comment]
Superceded by #136

**Pull request closed without merging**